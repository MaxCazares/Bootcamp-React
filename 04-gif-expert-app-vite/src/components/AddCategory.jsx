import { useState } from "react";

export const AddCategory = ({onNewCategory}) => {

    const [inputValue, setInputValue] = useState('')

    const onInputChange = ({target}) => setInputValue(target.value) 

    const onSubmit = event => {
        event.preventDefault()

        const newCategory = inputValue.trim()
        if(newCategory.length <= 1) return;

        onNewCategory(newCategory)
        setInputValue('')
    }

    return(
        <form onSubmit={onSubmit} className="flex justify-center mt-8">
            <input  
                className="w-11/12 max-w-2xl bg-purple-300 text-white placeholder-white rounded-lg px-4 py-2 shadow-md focus:outline-none focus:ring-2 focus:ring-purple-300 transition-all duration-200"
                type="text" 
                placeholder="Busca algo..."
                value={inputValue}
                onChange={onInputChange}
            />
        </form>
    )
}