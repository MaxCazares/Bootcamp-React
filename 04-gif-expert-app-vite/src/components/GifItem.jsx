import PropTypes from 'prop-types'

export const GifItem = ({title, url}) => {
    return(
        <div className="w-60 h-64 flex-col items-center justify-center bg-gray-100 p-1.5 m-2 rounded-xl shadow-2xl">
            <img
                className="w-full h-4/5 object-cover"
                src={url}
                alt={title}
            />
            <p className="w-full h-1/5 flex items-center justify-center text-center text-sm font-medium">
                {title}
            </p>
        </div>
    )
}

GifItem.propTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired,
}