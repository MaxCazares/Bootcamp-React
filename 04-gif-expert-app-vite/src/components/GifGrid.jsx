import React from 'react'
import { GifItem } from './GifItem'
import { useFetchGifs } from '../hooks/useFetchGifs'

export const GifGrid = ({ category }) => {

    const {images, isLoading} = useFetchGifs(category)
    
    return (
        <div className=''>
            <h1 className="text-left text-4xl font-bold py-5 px-8">{category}</h1>

            {isLoading && <h2>Cargando...</h2>}

            <div className="flex flex-wrap justify-center">
                {
                    images.map((image                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                   ) => (
                        <GifItem 
                            key={image.id} 
                            {...image}
                        />
                    ))
                }
            </div>
        </div>
    )
}
