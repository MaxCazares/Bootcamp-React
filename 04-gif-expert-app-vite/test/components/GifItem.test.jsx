import { render, screen } from "@testing-library/react"
import { GifItem } from "../../src/components/GifItem"

describe('Pruebas en <GifItem />', () => {
    
    const title = 'Vegeta'
    const url = 'https://dragon-ball/vegeta.jpg'
    
    test('Debe de hacer match con el snapshot', () => {
        const {container} = render(<GifItem title={title} url={url}/>)

        expect(container).toMatchSnapshot()

    })

    test('Debe de mostrar el alt y el url indicados', () => {
        render(<GifItem title={title} url={url}/>)
        // screen.debug()
        const {src, alt} = screen.getByRole('img')
        expect(src).toBe(url)
        expect(alt).toBe(title)        
    })

    test('Debe de mostar el titulo en el componente', () => {
        render(<GifItem title={title} url={url}/>)

        expect(screen.getByText(title)).toBeTruthy()
    })
})