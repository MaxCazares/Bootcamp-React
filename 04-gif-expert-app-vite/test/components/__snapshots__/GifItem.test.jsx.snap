// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Pruebas en <GifItem /> Debe de hacer match con el snapshot 1`] = `
<div>
  <div
    class="w-60 h-64 flex-col items-center justify-center bg-gray-100 p-1.5 m-2 rounded-xl shadow-2xl"
  >
    <img
      alt="Vegeta"
      class="w-full h-4/5 object-cover"
      src="https://dragon-ball/vegeta.jpg"
    />
    <p
      class="w-full h-1/5 flex items-center justify-center text-center text-sm font-medium"
    >
      Vegeta
    </p>
  </div>
</div>
`;
